src/realm/sync/noinst/client_reset.cpp

```diff
@@ -416,8 +432,12 @@ constexpr static std::string_view s_meta_reset_table_name("client_reset_metadata
 constexpr static std::string_view s_pk_col_name("id");
 constexpr static std::string_view s_version_column_name("version");
 constexpr static std::string_view s_timestamp_col_name("event_time");
-constexpr static std::string_view s_reset_type_col_name("type_of_reset");
-constexpr int64_t metadata_version = 1;
+constexpr static std::string_view s_reset_mode_col_name("type_of_reset");
+constexpr static std::string_view s_reset_recovery_col_name("reset_recovery_allowed");
+constexpr static std::string_view s_reset_action_col_name("reset_action");
+constexpr static std::string_view s_reset_error_code_col_name("reset_error_code");
+constexpr static std::string_view s_reset_error_msg_col_name("reset_error_msg");
+constexpr int64_t metadata_version = 2;
```

Comment:
Not your naming, but it should be:
```suggestion
constexpr int64_t s_metadata_version = 2;
```